server/src/test/java/org/elasticsearch/cluster/metadata/IndexMetadataTests.java

```diff
@@ -504,11 +505,21 @@ public void testIndexAndAliasWithSameName() {
                 .numberOfShards(1)
                 .numberOfReplicas(0)
                 .putAlias(AliasMetadata.builder("index").build())
-                .build()
+                .build(randomBoolean())
```

Comment:
Could we expand this test (or add one) to also verify that for indices created in v8.5, we reject an alias of the same name as the index?